lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/ToGenericMathLang.hs

```diff
@@ -100,6 +105,8 @@ insertTypeDecls l4i = fmap (insertIntoRule allTypeDecls)
     allTypeDecls :: Maybe ParamText
     allTypeDecls = tdRules |> foldMap rule2TypeDecls |> nonEmpty
 
+-- | TODO: Needs documentation.
+--
```

Comment:
I'll add this, do you think it's understandable?

```
-- This function expands the rules, i.e. inserting child rules into parent rules.
-- To do that, it calls expandClauses from Interpreter. Direct output of
-- expandClauses leaves the child rules intact, which results in redundancy.
-- So the line "r.name `notElem` leaves" removes the child rules that have
-- already been inserted into the parent.
```

Reply:
Re `expandClauses`: When I first saw what `expandClauses` months ago, my initial reaction / worry was that it's doing a lossy transformation, and that might pose issues for downstream consumers. Not sure if this is a real worry though. (Also, it may not be doing the substitution in an efficient way --- it doesn't seem to be using any sort of cache?)

(I wasn't following the discussion in the meeting today, so apologies if this was already discussed.)

Reply:
I'll add the documentation suggested by @inariksit for now and add a TODO about looking at `expandClauses`.